componentLibraries/defaultLibrary/Signal/Logic/SignalSteadyStateIdentifier.hpp

```diff
@@ -0,0 +1,217 @@
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+//!
+//! @file   SignalSteadyStateIdentifier.hpp
+//! @author Robert Braun <robert.braun@liu.se>
+//! @date   2022-03-10
+//!
+//! @brief Steady-state identifier component
+//!
+//$Id$
+
+#ifndef SIGNALSTEADYSTATEIDENTIFIER_H
+#define SIGNALSTEADYSTATEIDENTIFIER_H
+
+#include "ComponentEssentials.h"
+#include "ComponentUtilities.h"
+
+namespace hopsan {
+
+    //!
+    //! @brief
+    //! @ingroup SignalComponents
+    //!
+    class SignalSteadyStateIdentifier : public ComponentSignal
+    {
+
+    private:
+        double *mpIn, *mpOut, *mpTol, *mpWl, *mpSd, *mpL1, *mpL2, *mpL3;
+        int mMethod;
+        std::vector<double> mWindow;
+        size_t mWindowId;
+        double mDelayedX;
+        double mDelayedXf;
+        double mDelayedDf;
+
+    public:
+        static Component *Creator()
+        {
+            return new SignalSteadyStateIdentifier();
+        }
+
+        void configure()
+        {
+            addInputVariable("in", "", "", 0.0, &mpIn);
+            addOutputVariable("out", "1 if steadystate, else 0", "", &mpOut);
+
+            std::vector<HString> methods;
+            methods.push_back("1: Rectangular sliding window");
+            methods.push_back("2: Variance ratio test (F-test)");
+            methods.push_back("3: Exponentially weighted moving average (R-test)");
+
+            addConditionalConstant("method", "Estimation method", methods, 0, mMethod);
+
+            addInputVariable("tol", "Tolerance", "", 1.0, &mpTol);
+            addInputVariable("wl", "Length of sliding window (for methods 1 and 2)", "", 0.1, &mpWl);
+            addInputVariable("sd", "Standard deviation of white noise (for methods 2 and 3)", "", 0.01, &mpSd);
+            addInputVariable("l1", "Filter factor 1 (for method 3)", "", 0.1, &mpL1);
+            addInputVariable("l2", "Filter factor 2 (for method 3)", "", 0.1, &mpL2);
+            addInputVariable("l3", "Filter factor 3 (for method 3)", "", 0.1, &mpL3);
+        }
+
+
+        void initialize()
+        {
+            mWindowId = 0;
+            mWindow.resize(*mpWl/mTimestep, 0);
+
+            mDelayedX = (*mpIn);
+            mDelayedXf = (*mpIn);
+            mDelayedDf = 0;
+
+            if(mWindow.size() <= 2) {
+                stopSimulation("Sliding window is too small compared to time step");
+                return;
+            }
+
+            simulateOneTimestep();
+        }
+
+
+        void simulateOneTimestep()
+        {
+            if(mMethod == 0) {  //Rectangular sliding window
+                mWindow[mWindowId] = *mpIn;
+                mWindowId++;
+                if(mWindowId > mWindow.size()-1) {
+                    mWindowId = 0;
+                }
+
+                double xmax = mWindow[0];
+                double xmin = mWindow[0];
+                for(int i=0; i<mWindow.size(); ++i) {
```

Comment:
Risk for slow simulation, loops over entire window every timestep, maybe this is needed, or maybe this could be made some other way. Maybe update min/max value in window every time a new sample is added/ removed. 
But OK a see the problem, when a value slides out of the window, how should we then update min/max values.

Reply:
Yes, it would not work if the discarded value is the maximum. Perhaps we could optimize this by checking if the max/min needs to be updated or not, but it would risk making the code overcomplicated.